go-wasm-game/main.go

```diff
@@ -223,8 +215,14 @@ func recenterSquare(this js.Value, args []js.Value) interface{} {
 	mapWorldWidth := float64(gameMap.Width) * gameMap.TileSize
 	mapWorldHeight := float64(gameMap.Height) * gameMap.TileSize
 	
-	x = (mapWorldWidth - 20) / 2
-	y = (mapWorldHeight - 20) / 2
+	centerX := (mapWorldWidth - player.Width) / 2
+	centerY := (mapWorldHeight - player.Height) / 2
+	
+	player.X = centerX
```

Comment:
This and the following lines should be a player.position and player.settarget function